game-core/src/main/java/games/strategy/triplea/ai/proAI/data/ProPurchaseOption.java

```diff
@@ -246,9 +246,7 @@ public double getSeaDefenseEfficiency(final GameData data, final List<Unit> owne
     if (needDestroyer && isDestroyer) {
       seaFactor = 8;
     }
-    if (isAir) {
-      seaFactor = 4;
-    } else if (carrierCapacity > 0 && unusedLocalCarrierCapacity <= 0) {
+    if (isAir || carrierCapacity > 0 && unusedLocalCarrierCapacity <= 0) {
```

Comment:
You should definitely put the and in parentheses for better readability (also a thing that error-prone checks for)

Reply:
I've an auto-cleanup that will add them in. I was planning to run that once some of the code updates died down.